### PR TITLE
Anurse/bootstrap with nuget finally

### PR DESCRIPTION
**Pull request by piotrpMSFT**
Contributing to @anurse's work

### CODE REVIEW

File: scripts/compile/compile-stage.ps1 [hunk 1]
```diff
@@ -130,10 +130,10 @@ $BinariesForCoreHost | ForEach-Object {
 }
 
 # Crossgen Roslyn
-if (-not (Test-Path "$OutputDir\bin\csc.ni.exe")) {
-    header "Crossgening Roslyn compiler ..."
-    _cmd "$RepoRoot\scripts\crossgen\crossgen_roslyn.cmd ""$OutputDir"""
-}
+#if (-not (Test-Path "$OutputDir\bin\csc.ni.exe")) {
```
Comment (davidfowl) [inline scripts/compile/compile-stage.ps1]: What?

File: scripts/compile/compile-stage.sh [hunk 2]
```diff
@@ -115,10 +115,10 @@ cd $OUTPUT_DIR
 find . -type f | xargs chmod 644
 $REPOROOT/scripts/build/fix-mode-flags.sh
 
-if [ ! -f "$OUTPUT_DIR/bin/csc.ni.exe" ]; then
-    info "Crossgenning Roslyn compiler ..."
-    $REPOROOT/scripts/crossgen/crossgen_roslyn.sh "$OUTPUT_DIR/bin"
-fi
+#if [ ! -f "$OUTPUT_DIR/bin/csc.ni.exe" ]; then
+    #info "Crossgenning Roslyn compiler ..."
```
Comment (davidfowl) [inline scripts/compile/compile-stage.sh]: ??

**Comment by piotrpMSFT** [comment]
@davidfowl we were seeing crashes in xgen so turned it off for the moment to get this moving.

File: src/Microsoft.DotNet.ProjectModel.Loader/project.json [hunk 3]
```diff
@@ -4,11 +4,13 @@
...
-        "NETStandard.Library": "1.0.0-rc2-23704",
+        "NETStandard.Library": "1.0.0-rc3-23722",
         "Microsoft.DotNet.ProjectModel": "1.0.0-*",
-        "System.Runtime.Loader": "4.0.0-rc2-23704"
+        "System.Runtime.Loader": "4.0.0-rc3-23722"
     },
     "frameworks": {
-        "dnxcore50": { }
+        "dnxcore50": {
+            "imports": "portable-net451+win8"
```
Comment (davidfowl) [inline src/Microsoft.DotNet.ProjectModel.Loader/project.json]: Why is this required?

**Comment by piotrpMSFT** [comment]
OS X failures were due to https://github.com/NuGet/Home/issues/1977

Thanks, @davidfowl for tracking that down and providing a workaround.

**Comment by anurse** [comment]
> 11:40:52 error: System.Security.Principal.Windows 4.0.0-rc2-23722 provides a compile-time reference assembly for System.Security.Principal.Windows on DNXCore,Version=v5.0, but there is no run-time assembly compatible with centos.7-x64.

We're getting the broken packages again. /cc @ericstj

**Comment by piotrpMSFT** [comment]
yea, just fixed it. Somehow VS Code's filesystemwatcher stopped paying attention to my sln dir and search did not find this one. I'm going to recycle the process and make sure I don't find others.

**Comment by ericstj** [comment]
@anurse **rc3**-23722 has the fix.  **rc2**-23722 is the rel branch which did not have the fix yet.  For rel branch we won't get the fix until tomorrow: https://github.com/dotnet/corefx/pull/5629.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu Debug Build please.

**Comment by anurse** [comment]
@ericstj Yeah, I pretty much knew that, just confirming.

**Comment by anurse** [comment]
@dotnet-bot test Ubuntu Debug Build please

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu Debug Build please.

**Comment by anurse** [comment]
Since the Ubuntu Debug build keeps failing, I wonder if something is corrupt in the NuGet HTTP cache causing this error to reoccur. Maybe we should just write the Debug build off for now and look at cleaning the caches pre-build.

**Comment by anurse** [comment]
Windows build failures are caused by the fact that we are getting a pre-crossgenned `mscorlib` out of `Microsoft.NETCore.Runtime` and it is affected by the crossgen bug.

**Comment by piotrpMSFT** [comment]
https://github.com/dotnet/coreclr/issues/2850 is tracking.

@gkhanna79 please note this is the only remaining Beta blocker for us.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu Debug Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu Release Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Windows_NT Debug Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test this please.

**Comment by piotrpMSFT** [comment]
re-running, there should be a NuGet fix out which we'll pick up during restore which will address our Ubuntu issues.

**Comment by brthor** [comment]
@piotrpMSFT Thinking about this, we will need that fix in the stage0 to take advantage of it.

**Comment by piotrpMSFT** [comment]
True, but note that this failure has a stack trace. I'll make a PR from home to pull in 531, unless you have cycles to do this.

**Comment by brthor** [comment]
@Sridhar-MS you found out something about this stack trace?

**Comment by Sridhar-MS** [comment]
I do have a theory regarding the ubuntu restore failure. corefx is currently publishing packages into the myget feed - https://www.myget.org/gallery/dotnet-core. And all the nuget restore error `Failed to download package from` seem to happen only on dotnet-core. So this could be related.

http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/debug_ubuntu_prtest/983/console

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu Debug Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test this please.

**Comment by anurse** [comment]
The Windows failures remain the Crossgen issue? Now that we've got a successful Ubuntu build (the Debug one), I suggest we call Ubuntu good enough :). Of course if we need to integrate the crossgen fix, we'll need to spin another build anyway.

**Comment by piotrpMSFT** [comment]
@anurse for Windows we're waiting on https://github.com/dotnet/coreclr/pull/2853. @Sridhar-MS has confirmed that with this fix he was able to do multiple full Windows builds without the crash.

For Ubuntu we are investigating why NuGet v3 is much slower on this platform as well as returning non-0 so frequently. Agreed, however, that we may treat this as a non-blocking issue.

**Comment by anurse** [comment]
:+1:, just anxious to get this in. I'm sure I'm the **only** one feeling like that ;P

**Comment by piotrpMSFT** [comment]
hehe, yea. But if we don't have passing CI we'll be stuck in a world where we have a broken stage 0... that'd be hard to get out of.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test CentOS7.1 Debug Build please.

**Comment by piotrpMSFT** [comment]
@anurse 23727 hasn't dropped yet... but all other issues are addressed :)

**Comment by piotrpMSFT** [comment]
@anurse this should be it...

**Comment by anurse** [comment]
\*crosses fingers*

**Comment by piotrpMSFT** [comment]
* preps appropriate GIF *

**Comment by piotrpMSFT** [comment]
One Green...

**Comment by piotrpMSFT** [comment]
Two green...

**Comment by piotrpMSFT** [comment]
vader

**Comment by piotrpMSFT** [comment]
@anurse wohoo!!!

**Comment by brthor** [comment]
NICE
:+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1:
